qiskit_dynamics/pulse/pulse_simulator.py

```diff
@@ -0,0 +1,330 @@
+            description='Pulse enabled simulator backend.',
+            backend_version=0.1
+        )
+
+        self._solver = solver
+        self._subsystem_dims = subsystem_dims or [solver.model.dim]
+        self._subsystem_labels = subsystem_labels or list(range(len(subsystem_dims)))
+
+        # get the dressed states
+        static_hamiltonian = _get_lab_frame_static_hamiltonian(solver.model)
+        dressed_evals, dressed_states = _get_dressed_state_decomposition(static_hamiltonian)
+        self._dressed_evals = dressed_evals
+        self._dressed_states = dressed_states
+        self._dressed_states_adjoint = self._dressed_states.conj().transpose()
+
+    @property
+    def solver(self):
+        """The internal Solver."""
+        return self._solver
+
+    @property
+    def subsystem_dims(self):
+        """Subsystem dimensions."""
+        return self._subsystem_dims
+
+    @property
+    def subsystem_labels(self):
+        """Subsystem labels."""
+        return self._subsystem_labels
+
+    def run(
+        self,
+        run_input: Union[Schedule, ScheduleBlock],
+        shots: int = 1,
+        y0 = None,
+        validate: Optional[bool] = False,
+        normalize_states: Optional[bool] = True,
+        solver_options: Optional[dict] = None,
+        **options
+    ) -> Result:
+        """Run on the backend.
+
+        normalize_states ensures that, when sampling, the state is normalized, to avoid errors
+        being raised due to numerical tolerance issues.
+
+        Notes/questions:
+        - Should we force y0 to be a quantum_info state? This currently assumes that
+        - Should we provide optional arguments to run that allow the user to specify different
+          modes of simulation? E.g.
+            - Just simulate the state
+            - Simulate the unitary or process
+            - Return probabilities
+            - Return counts
+          For now, assuming just counts
+        - Validate which channels are available on device before running? Can the BackendV2 be
+          set up to do this automatically somehow?
+        - What is the formatting for the results when measuring a subset of qubits? E.g. if
+          you measure qubits [0, 2, 3], do you get counts for 3-bit strings out?
+        - Add validation that only qubits present in the model are being measured. This may
+          happen automatically as the measurement code will fail, but would be good to raise
+          a proper error.
+        - How to handle binning of higher level measurements? E.g. should everything above 0
+          count as 1?
+        - What to do with memory and register slots?
+
+        To test:
+        - Measuring a subset of qubits when more than one present
+        - If t_span=[0, 0] it seems that NaNs appear for JAX simulation - should solve this.
+        - Normalizing the state.
+        """
+
+        if validate:
+            _validate_run_input(run_input)
+
+        if solver_options is None:
```

Comment:
Options should be obtained from current values in self.options, and then overridden with any passed runtime options